### PR TITLE
1-st pull request

### PR DESCRIPTION
**Pull request by volodymyrtaras**
1.1. I have done commands git merge, git rebase, git cherry-pick
1.2. I have done the task

### CODE REVIEW

File: src/lesson/third/car/Car.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    private final Calendar date;
+    private String typeOfEngine;
+    private int maxSpeed;
+    private double fromZeroToHundred;
+    private int amountOfPassengers;
+    private int currentPassengers;
+    private int currentMaxSpeed;
+    private double[] wheels;
+    private boolean[] doors;
+
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: You should add constructor without parameters.

Comment (volodymyrtaras) [reply src/lesson/third/car/Car.java]: @Anderwerty the task is "Нет пустого конструктора. Так как есть поля в классе, которые нельзя изменять после создания объекта. Например дата производства.". Didn't I understand it clearly?

Comment (Anderwerty) [reply src/lesson/third/car/Car.java]: Example:
```
public Car() {
        this.maxSpeed =100;
    }
```

File: src/lesson/third/car/Car.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        System.out.println("Type of engine is: " + getTypeOfEngine());
+        System.out.println("Maximum speed is: " + getMaxSpeed() + "km/h");
+        System.out.println("From zero to a hundred accelerates in: " + getFromZeroToHundred() + "s.");
+        System.out.println("Number of passengers is: " + getAmountOfPassengers());
+        System.out.println("Take a look on one of the wheels: " + getWheelFromIndex(2));
+        System.out.println("Take a look on one of the doors: " + getDoorFromIndex(1));
+        System.out.println("Current passengers " + getCurrentPassengers() + " passengers");
+        System.out.println("Current maximum speed is: " + getCurrentMaxSpeed() + "km/h");
+    }
+}
```
[the remark this answers is not in the document]
Comment (volodymyrtaras) [reply src/lesson/third/car/Car.java]: @Anderwerty blank line between sout and }}?

[the remark this answers is not in the document]
Comment (Anderwerty) [reply src/lesson/third/car/Car.java]: after
```
    }
}
```
image

**Comment by volodymyrtaras** [comment]
@Anderwerty fixed.

File: src/lesson/third/car/Car.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    public boolean[] getDoors() {
+        return doors;
+    }
+
+    //constructors
+    public Car(LocalDate date) {
+        this.date = date;
+    }
+
+    public Car() {
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: Order: fields, constructors,  setter and/or getters and after all other methods.

File: src/lesson/third/car/Car.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        this.typeOfEngine = "V6";
+        this.maxSpeed = 100;
+        this.fromZeroToHundred = 7.5;
+        this.amountOfPassengers = 5;
+        this.currentPassengers = 3;
+        this.currentMaxSpeed = 80;
+        this.wheels = new double[]{new CarWheel().getWheelState(), new CarWheel().getWheelState(), new CarWheel().getWheelState(), new CarWheel().getWheelState()};
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: Very long line.

File: src/lesson/third/car/Car.java [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        this.doors = new boolean[]{new CarDoor(false, false).isDoorIsOpen(), new CarDoor(false, false).isWindowIsOpen(), new CarDoor(false, false).isDoorIsOpen(), new CarDoor(false, false).isWindowIsOpen()};
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: The same.

File: src/lesson/third/car/Car.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        this.date = null;
+    }
+
+    public Car(LocalDate date, String typeOfEngine, int maxSpeed, double fromZeroToHundred, int amountOfPassengers, int currentPassengers, int currentMaxSpeed) {
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: The same

File: src/lesson/third/car/Car.java [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        this.date = date;
+        this.typeOfEngine = typeOfEngine;
+        this.maxSpeed = maxSpeed;
+        this.fromZeroToHundred = fromZeroToHundred;
+        this.amountOfPassengers = amountOfPassengers;
+        this.currentPassengers = currentPassengers;
+        this.currentMaxSpeed = currentMaxSpeed;
+        this.doors = new boolean[]{new CarDoor(false, false).isDoorIsOpen(), new CarDoor(false, false).isWindowIsOpen(), new CarDoor(false, false).isDoorIsOpen(), new CarDoor(false, false).isWindowIsOpen()};
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: The same.

File: src/lesson/third/car/Car.java [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    //other methods
+    public void changeCurrentSpeed(int currentSpeed) {
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: Do we need this comment?

File: src/lesson/third/car/Car.java [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+            this.currentPassengers = 0;
+        }
+    }
+
+    public void clearTheCar() {
+        this.currentPassengers = 0;
+    }
+
+    public boolean getDoorFromIndex(int index) {
+        if (index < 0) {
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: index<0 || index>getDoors().length

File: src/lesson/third/car/Car.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Anderwerty) [reply src/lesson/third/car/Car.java]: Add a blank line.

**Comment by volodymyrtaras** [comment]
@Anderwerty fixed.

File: src/lesson/third/car/Car.java [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+        Arrays.sort(getWheels());
+        currentMaxSpeed = maxSpeed * (int) (this.wheels[this.wheels.length - 1]);
+
+        if (currentPassengers == 0) {
+            currentMaxSpeed = 0;
+        }
+    }
+
+    @Override
+    public String toString() {
```
Comment (Anderwerty) [inline src/lesson/third/car/Car.java]: No. toString() should return this text. And another method should print it.

File: src/lesson/third/car/CarDoor.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    public String printDoor( boolean isDoorTrue) {
+        return isDoorTrue ? "The door is open" : "The door is closed";
+    }
+
+    public String printWindow( boolean isWindowTrue) {
+        return isWindowTrue ? "The window is open" : "The window is closed";
+    }
+}
+
+
```
Comment (Anderwerty) [inline src/lesson/third/car/CarDoor.java]: One empty line enough.

File: src/lesson/third/car/CarWheel.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "Current state of the car wheel is: " + getWheelState();
+    }
+
+}
+
+
```
Comment (Anderwerty) [inline src/lesson/third/car/CarWheel.java]: One empty line enough.

File: src/lesson/third/car/Start.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+        car.clearTheCar();
+        car.getInTheCar();
+        car.takeOffWheels();
+        car.addWheels(4);
+        car.currentMaxSpeed();
+        car.toString();
+    }
+}
+
+
```
Comment (Anderwerty) [inline src/lesson/third/car/Start.java]: One empty line enough.

File: out/production/HomeworkMateAcademy/lesson/third/WhatToDoThree [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+* Получить колесо по индексу
+* Снять все колеса с машины
+* Установить на машину X новых колесу (вдобавок к имеющимся, то есть если было 4 колеса, после вызова метода с Х аргументом равным трем, колес будет 4+3=7)
+* Вычислить текущую возможную максимальную скорость (Скорость машины вычисляется так. Максимальная скорость новой машины множиться на самое стертое колесо в машине. Максимальная скорость равна 0 если в машине нет ни одного пассажира, так как некому ее вести)
+* Вывести в консоль данные об объекте (все поля и вычисленную максимальную скорость в зависимости от целостности колес и наличия водителя)
+
+
+Задание 2 (дополнительное)
+Создать консольный пользовательский интерфейс. В котором пользователя программа будет спрашивать какое действие выполнить и с какими параметрами.
+Кол-во различных действий = кол-ву функций в ДЗ.
```
Comment (Anderwerty) [inline out/production/HomeworkMateAcademy/lesson/third/WhatToDoThree]: Add a blank line.

**Comment by volodymyrtaras** [comment]
@Anderwerty fixed.

**Comment by volodymyrtaras** [comment]
@Anderwerty fixed.

File: src/lesson/third/car/CarDoor.java [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+        this.windowIsOpen = !windowIsOpen;
+    }
+
+    public void printCarDoor(boolean isDoorTrue, boolean isWindowTrue) {
+        printDoor(isDoorTrue);
+        printWindow(isWindowTrue);
+    }
+
+    public String printDoor(boolean isDoorTrue) {
+        return isDoorTrue ? "The door is open" : "The door is closed";
```
Comment (Anderwerty) [inline src/lesson/third/car/CarDoor.java]: Please use constants for "The door is open"  and "The door is closed.

File: src/lesson/third/car/CarDoor.java [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public String printWindow(boolean isWindowTrue) {
+        return isWindowTrue ? "The window is open" : "The window is closed";
```
Comment (Anderwerty) [inline src/lesson/third/car/CarDoor.java]: The same.

**Comment by volodymyrtaras** [comment]
@Anderwerty thank you))

**Pull request merged**